src/metalperformanceshadersgraph.cs

```diff
@@ -0,0 +1,1840 @@
+
+		[Export ("runWithMTLCommandQueue:feeds:targetTensors:targetOperations:")]
+		MPSGraphTensorDataDictionary Run (IMTLCommandQueue commandQueue, MPSGraphTensorDataDictionary feeds, MPSGraphTensor[] targetTensors, [NullAllowed] MPSGraphOperation[] targetOperations);
+
+		[Export ("runWithMTLCommandQueue:feeds:targetOperations:resultsDictionary:")]
+		void Run (IMTLCommandQueue commandQueue, MPSGraphTensorDataDictionary feeds, [NullAllowed] MPSGraphOperation[] targetOperations, MPSGraphTensorDataDictionary resultsDictionary);
+
+		[Export ("runAsyncWithFeeds:targetTensors:targetOperations:executionDescriptor:")]
+		MPSGraphTensorDataDictionary RunAsync (MPSGraphTensorDataDictionary feeds, MPSGraphTensor[] targetTensors, [NullAllowed] MPSGraphOperation[] targetOperations, [NullAllowed] MPSGraphExecutionDescriptor executionDescriptor);
+
+		[Export ("runAsyncWithMTLCommandQueue:feeds:targetTensors:targetOperations:executionDescriptor:")]
+		MPSGraphTensorDataDictionary RunAsync (IMTLCommandQueue commandQueue, MPSGraphTensorDataDictionary feeds, MPSGraphTensor[] targetTensors, [NullAllowed] MPSGraphOperation[] targetOperations, [NullAllowed] MPSGraphExecutionDescriptor executionDescriptor);
+
+		[Export ("runAsyncWithMTLCommandQueue:feeds:targetOperations:resultsDictionary:executionDescriptor:")]
+		void RunAsync (IMTLCommandQueue commandQueue, MPSGraphTensorDataDictionary feeds, [NullAllowed] MPSGraphOperation[] targetOperations, MPSGraphTensorDataDictionary resultsDictionary, [NullAllowed] MPSGraphExecutionDescriptor executionDescriptor);
+
+		[Export ("encodeToCommandBuffer:feeds:targetTensors:targetOperations:executionDescriptor:")]
+		MPSGraphTensorDataDictionary Encode (MPSCommandBuffer commandBuffer, MPSGraphTensorDataDictionary feeds, MPSGraphTensor[] targetTensors, [NullAllowed] MPSGraphOperation[] targetOperations, [NullAllowed] MPSGraphExecutionDescriptor executionDescriptor);
+
+		[Export ("encodeToCommandBuffer:feeds:targetOperations:resultsDictionary:executionDescriptor:")]
+		void Encode (MPSCommandBuffer commandBuffer, MPSGraphTensorDataDictionary feeds, [NullAllowed] MPSGraphOperation[] targetOperations, MPSGraphTensorDataDictionary resultsDictionary, [NullAllowed] MPSGraphExecutionDescriptor executionDescriptor);
+
+	}
+
+	// @interface MPSGraphGradientOps (MPSGraph)
+	[iOS (14,0), TV (14,0), Mac (11,0), MacCatalyst (14,0)]
+	[Category]
+	[BaseType (typeof(MPSGraph))]
+	interface MPSGraph_MPSGraphGradientOps
+	{
+		// -(NSDictionary<MPSGraphTensor *,MPSGraphTensor *> * _Nonnull)gradientForPrimaryTensor:(MPSGraphTensor * _Nonnull)primaryTensor withTensors:(NSArray<MPSGraphTensor *> * _Nonnull)tensors name:(NSString * _Nullable)name __attribute__((swift_name("gradients(of:with:name:)")));
+		[Export ("gradientForPrimaryTensor:withTensors:name:")]
+		NSDictionary<MPSGraphTensor, MPSGraphTensor> Gradients (MPSGraphTensor of, MPSGraphTensor[] with, [NullAllowed] string name);
```

Comment:
Maybe Get as a verb here? since it is a method

```suggestion
		NSDictionary<MPSGraphTensor, MPSGraphTensor> GetGradients (MPSGraphTensor of, MPSGraphTensor[] with, [NullAllowed] string name);
```

Reply:
I don't think so since it breaks the style. The ops read as declarative statements - nouns - not verbs. Though this is a complex op, it's still just an op. Also, I prefer to match the swift name for sanity. :-)